src/nrnpython/nrnpy_nrn.cpp

```diff
@@ -1365,26 +1364,25 @@ static PyObject* NPyMechObj_is_ion_safe(NPyMechObj* self) {
 
 static PyObject* NPyMechObj_segment(NPyMechObj* self) {
     CHECK_PROP_INVALID(self->prop_id_);
-    PyObject* result = NULL;
-    if (self->pyseg_) {
-        result = (PyObject*) (self->pyseg_);
-        Py_INCREF(result);
-    }
-    return result;
+    auto pyseg = nb::borrow((PyObject*) self->pyseg_);
+    return pyseg.release().ptr();
 }
 
 static PyObject* NPyMechObj_segment_safe(NPyMechObj* self) {
     return nrn::convert_cxx_exceptions(NPyMechObj_segment, self);
 }
 
-static PyObject* NPyMechFunc_mech(NPyMechFunc* self) {
-    PyObject* result = NULL;
-    if (self->pymech_) {
-        CHECK_PROP_INVALID(self->pymech_->prop_id_);
```

Comment:
@nrnhines We have this `CHECK_PROP_INVALID` in `NPyMechFunc_mech` but not in `NPyRangeVar_mech` even though the code looked very similar. 
Is this intentional? If both could be consistent we can use a c++template to have a single version of the code (see suggested change)